piet-cairo/src/lib.rs

```diff
@@ -68,15 +68,33 @@ impl<'a> RenderContext for CairoRenderContext<'a> {
         Ok(())
     }
 
-    fn clear(&mut self, _region: impl Into<Option<Rect>>, color: Color) {
-        let rgba = color.as_rgba_u32();
-        self.ctx.set_source_rgba(
-            byte_to_frac(rgba >> 24),
-            byte_to_frac(rgba >> 16),
-            byte_to_frac(rgba >> 8),
-            byte_to_frac(rgba),
-        );
-        self.ctx.paint();
+    fn clear(&mut self, region: impl Into<Option<Rect>>, color: Color) {
+        let _ = self.with_save(|rc| {
+            rc.ctx.reset_clip();
+            let matrix = affine_to_matrix(rc.current_transform());
+            rc.transform(rc.current_transform().inverse());
+
+            //prepare the colors etc
+            let rgba = color.as_rgba_u32();
+            rc.ctx.set_source_rgba(
+                byte_to_frac(rgba >> 24),
+                byte_to_frac(rgba >> 16),
+                byte_to_frac(rgba >> 8),
+                byte_to_frac(rgba),
+            );
+            // we DO want to clip the specified region
+            if let Some(region) = region.into() {
+                rc.clip(region)
+            }
+            //we also need to save/restore the operator, so we can apply the SOURCE operator
+            let op = rc.ctx.get_operator();
+            rc.ctx.set_operator(cairo::Operator::Source);
+            rc.ctx.paint();
+            rc.ctx.set_operator(op);
```

Comment:
I'm not *sure* we need to do this, since the old operator might be saved and restored with the `save` operation? I thought I was testing this in the sample picture (by doing a fill after doing a clear) but I should double check that.

Reply:
yes i removed these in the mean time, thanks for pointing them out anyways!